rlog.go

```diff
@@ -626,6 +627,33 @@ func basicLog(logLevel int, traceLevel int, isLocked bool, format string, prefix
 	}
 }
 
+// fmtWithGID is like fmt.Sprintf, but adds another format character,
+// %G, which will be subsituted with Goroutine ID.
+func fmtWithGID(s string, args ...interface{}) string {
+	newStr := ""
+	for i := 0; i < len(s); i++ {
+		if i < len(s)-1 && s[i] == '%' && s[i+1] == 'G' {
```

Comment:
Does Go not have a 'find' function for substrings?

Also, it seems as if this assembles the new logline character by character. That seems to be pretty CPU intensive. Could we just check where %G appears for the first time, copy ALL the string so far over to the new string, then add the GID, then see where it appears again (if at all).

That way, loglines without %G (most of them for most users?) will be copied with a single operation after we perform a quick scan...

Reply:
Probably even better to use string built-ins. Fixing.